src/api/lexer/lexing.rs

```diff
@@ -0,0 +1,83 @@
+// This file is part of rust-web/twig
+//
+// For the copyright and license information, please view the LICENSE
+// file that was distributed with this source code.
+
+use api::Position;
+use api::tokens::TokenRef;
+
+use api::lexer::{ Lexer, LexingResult };
+use api::error::{ SyntaxErrorCode, UnexpectedToken };
+
+/// TokenRef wrapper for `Lexer` that additionaly has position.
+#[derive(Debug)]
+pub struct ItemRef<'t> {
+    pub token: TokenRef<'t>,
+    pub position: Position,
+}
+
+/// Lexer token iterator.
+///
+/// 'i is iteration lifetime, or "one use of lexer".
+/// 't is template lifetime. It will live longer than this iteration.
+pub struct Tokens<'i, 't> {
```

Comment:
`Tokens`: It's only a detail - I prefer singular names like "Team" instead of "People", or "String" instead of "Characters",. For example `lexer::Job` would still work in this case - I think it would reflect very well that it is some intermediate thing that has a mutable internal state. We could also call it `lexer::Iter`. :-)